### DISCUSSION TITLE
setting up radicale (carddav / caldav) with alpine 3.18 template

### DISCUSSION THREAD

**Discussion by tomtom1964bo** [tteck/Proxmox#2184] [Ideas]
Hi, if anyone is interested in a lightweight carddav server. Here are my steps for installing radicale on alpine 3.18.
For me it worked this way.

**1a - create new CT with alpine3.18 - Template (Disk 0,25GB , Ram 128MB, Swap 128MB)**

**1 - update and upgrade after install**
`apk update && apk upgrade`

**2 - Install radicale, nano and apache2-utils**
`apk add radicale nano apache2-utils`

**3 - Security: The storage should not be readable by others. (Run chmod -R o= /var/lib/radicale/collections as root.)**
`chmod -R o= /var/lib/radicale/collections`

**4 - Get your ip address**
`ip addr`

**5 - Edit /etc/radicale/config for setting ip address and authentification method**
`nano /etc/radicale/config`

```
[server]

# CalDAV server hostnames separated by a comma
# IPv4 syntax: address:port
# IPv6 syntax: [address]:port
# For example: 0.0.0.0:9999, [::]:9999
# Set your ip address
hosts = 192.168.1.196:5232

[auth]

# Authentication method
# Value: none | htpasswd | remote_user | http_x_remote_user
# Set htpasswd
type = htpasswd

# Htpasswd filename
# Set filename
htpasswd_filename = /etc/radicale/users

# Htpasswd encryption method
# Value: plain | bcrypt | md5
# bcrypt requires the installation of radicale[bcrypt].
# Set encryption method
htpasswd_encryption = bcrypt
```

**6 - Save the file**

**7 - Create user and password**
`htpasswd -B -c /etc/radicale/users user1`

**8 - Start service**
`rc-service radicale start`

**9 - Add service to default runlevel**
`rc-update add radicale`

**10 - webbrowser login**
`192.168.1.196:5232`

**Comment by knario** [comment]
Muchas gracias !!! estaba buscando como instalar radicale en un lxc.

**Reply by tomtom1964bo** [reply]
glad it helped you :-)

**Reply by knario** [reply]
Hola tomtom1964bo :
He tenido que borrar algunos lxc de mi servidor, por haber cambiado la configuración de proxmox.
Ahora al intentar instalar radicale con tu tutorial me da este error.

![Captura de Pantalla 2024-02-19 a las 11 15 09](https://github.com/tteck/Proxmox/assets/48197666/9a57ae76-8e08-4e87-a7b4-7d3c851ee1c7)


Lo he intentando con varias versiones de alpine y sigue el mismo error.


Mi archivo de configuración tiene algo así:

```
[server]

# CalDAV server hostnames separated by a comma
# IPv4 syntax: address:port
# IPv6 syntax: [address]:port
# For example: 0.0.0.0:9999, [::]:9999
hosts = 192.168.88.122:5232

[auth]

# Authentication method
# Value: none | htpasswd | remote_user | http_x_remote_user
#type = htpasswd

# Htpasswd filename
htpasswd_filename = /etc/radicale/users

# Htpasswd encryption method
# Value: plain | bcrypt | md5
# bcrypt requires the installation of radicale[bcrypt].
htpasswd_encryption = bcrypt


```

**Reply by tomtom1964bo** [reply]
Hola knario,
i'm no technician. But did you try to clear your browser cache? That is one possible solution i found on google.
Did you change your password? Some people had trouble with bcrypt-encryption.

**Reply by knario** [reply]
Muchas gracias por responder  tomtom1964bo !!!
Lo probaré sin Bcrypt-encryption, seguramente sea eso.
Feliz comienzo de semana. :-)

**Comment by Sciddie** [comment]
In my case radicale didn't work with this setup and gave a **500 Error.**
To fix this I had to **remove the command user line from /etc/init.d/radicale**

**Reply by tomtom1964bo** [reply]
Hi Florian,
thanks for your solution.

**Comment by asjohem** [comment]
Thanks, works like a charm! 

How do you remove a user, and how to change password for a user?
